### PR TITLE
S1/remaining fields

### PR DESCRIPTION
**Pull request by graemehoffman**
Build out of select and taxonomy fields.

### CODE REVIEW

File: app/Post_Types/Archive_Item/Post_Meta.php [hunk 1]
```diff
@@ -129,21 +129,42 @@ public function get_class( $post ) {
...
+			'fields' => 'ids',
+			'post_type' => Diviner_Field::NAME,
+			'meta_query' => $meta_query
+		);
+		$posts_ids = get_posts($args);
+
+		foreach($posts_ids as $post_id) {
+			$field_type = carbon_get_post_meta($post_id, FieldPostMeta::FIELD_TYPE, 'carbon_fields_container_field_variables');
+			$type = Diviner_Field::get_class( $field_type );
+			// call_user_func(array($field, 'setup'), $post_id);
```
Comment (lucatume) [inline app/Post_Types/Archive_Item/Post_Meta.php]: remove this comment

File: app/Post_Types/Diviner_Field/PostMeta.php [hunk 2]
```diff
@@ -98,8 +109,61 @@ public function add_post_meta() {
...
+	}
+
+	public function get_field_select_options() {
+		return Field::make( 'complex', self::FIELD_SELECT_OPTIONS )
+			->add_fields( array(
+				Field::make( 'text', self::FIELD_SELECT_OPTIONS_LABEL, 'Drop down label' ),
+			) );
+	}
+
+	public function get_field_CPT_ID() {
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/PostMeta.php]: why this mix of upper and lower case?

Comment (lucatume) [inline app/Post_Types/Diviner_Field/PostMeta.php]: why this mix, here and below, of upper and lower case?

File: app/Post_Types/Diviner_Field/Types/Select_Field.php [hunk 3]
```diff
@@ -12,8 +13,16 @@ class Select_Field extends FieldType {
...
 	const TYPE = 'select';
 
-	static public function render( $id, $field_label, $helper = '') {
+	static public function render( $post_id, $id, $field_label, $helper = '') {
+		$options = carbon_get_post_meta( $post_id, FieldPostMeta::FIELD_SELECT_OPTIONS);
+		if (count($options) <= 1) {
+			return '';
+		}
 		$field =  Field::make( static::TYPE, $id, $field_label );
+		$filtered_options = array_map( function ($a) {
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/Types/Select_Field.php]: use `array_column` if we're going for PHP 5.5+ compat or `wp_list_pluck` for compat w/ anything

File: app/Post_Types/Diviner_Field/Types/Taxonomy_Field.php [hunk 4]
```diff
@@ -5,20 +5,83 @@
...
+		$field_slug = carbon_get_post_meta( $post_id, FieldPostMeta::FIELD_TAXONOMY_SLUG);
+		$field_tax_type = carbon_get_post_meta( $post_id, FieldPostMeta::FIELD_TAXONOMY_TYPE);
+
+		if ( empty( $field_slug ) ) {
+			$field_slug = sanitize_title($field_label_singular);
+		}
+
+		if ( empty( $field_label_singular ) ) {
+			$field_label_singular = sprintf(
+				'Taxonomy %d',
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/Types/Taxonomy_Field.php]: Here and below: internationalisation.

File: app/Post_Types/Diviner_Field/Types/Taxonomy_Field.php [hunk 5]
```diff
@@ -5,20 +5,83 @@
...
+				'Taxonomies %d',
+				$post_id
+			);
+		}
+
+		// Labels
+		$labels = array(
+			'name'              => $field_label_plural,
+			'singular_name'     => $field_label_singular,
+			'search_items'      => sprintf( 'Search %s' , $field_label_plural ),
```
Comment (lucatume) [inline app/Post_Types/Diviner_Field/Types/Taxonomy_Field.php]: i18n

File: app/Post_Types/Archive_Item/Post_Meta.php (hunk 1 above)
Comment (graemehoffman) [reply app/Post_Types/Archive_Item/Post_Meta.php]: donen

File: app/Post_Types/Diviner_Field/PostMeta.php (hunk 2 above)
Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/PostMeta.php]: fixed

Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/PostMeta.php]: fixed

File: app/Post_Types/Diviner_Field/Types/Select_Field.php (hunk 3 above)
Comment (graemehoffman) [reply app/Post_Types/Diviner_Field/Types/Select_Field.php]: done

**Pull request merged**